python/lsst/ts/atmonochromator/monochromator_csc.py

```diff
@@ -455,6 +466,11 @@ async def do_updateMonochromatorSetup(
             if reply != ModelReply.OK:
                 raise RuntimeError(f"Got {reply} from controller.")
             else:
+                await self.cmd_updateMonochromatorSetup.ack_in_progress(
+                    data=data,
+                    timeout=self.model.move_grating_timeout,
```

Comment:
I think you need to send the in progress ack before you call the `self.model.set_all` no? Or does the call to `set_all` returns immediately?

Also , I think you might want to increase the timeout since it is waiting for more than one thing to finish. But I am not sure if this is done in series or in parallel, so your timeout might be ok.

Reply:
We want to do it before the wait_ready. The command returns an`#OK` first. I think the grating_timeout is long enough for all motions to complete since the grating takes the longest.